bpf-loader/src/eunomia_meta.cpp

```diff
@@ -182,6 +183,22 @@ bpf_skeleton::open_from_json_config(
     }
 }
 
+int
+bpf_skeleton::open_with_btf(const char *btf_path,
+                             std::vector<char> bpf_object_buffer) noexcept
+{
+    try {
+        state = ebpf_program_state::INIT;
+        __bpf_object_buffer = bpf_object_buffer;
+        custom_btf_path = get_btf_path(btf_path);
+        return 0;
+    } catch (std::runtime_error &e) {
+        std::cerr << "failed to get btf file " << e.what() << std::endl;
+        state = ebpf_program_state::INVALID;
+        return -1;
+    }
+}
```

Comment:
I don't think it would be a good idea to merge this function with `open_with_json_config`, since the former requires `btf_path` and the latter requires `json_data`.

Reply:
please write a test for this function.